rmgpy/reaction.py

```diff
@@ -1144,6 +1144,121 @@ def ensure_species(self, reactant_resonance=False, product_resonance=True):
+                except ValueError:
+                    continue
+                if not self.kinetics.isPressureDependent():
+                    kr_list.append(
+                        self.generateReverseRateCoefficient().getRateCoefficient(condition[0], condition[1]))
+            if self.kinetics.isPressureDependent():
+                for condition in pdep_conditions:
+                    kr_list.append(
+                        self.generateReverseRateCoefficient().getRateCoefficient(condition[0], condition[1]))
+                if len(pdep_conditions) > len(conditions):
+                    collision_limit = collision_limit * 2
+            for i, k in enumerate(kf_list):
+                if k > collision_limit[i]:
+                    r_violator = [self, 'reverse']
+                    logging.debug("Reaction {0} violates the collision rate limit in the"
+                                  " reverse direction!".format(self))
+                    break
+        return f_violator, r_violator
+
+    def calculate_coll_limit(self, temp, reverse=False):
+        """
+        Calculate the collision limit rate for the given temperature
+        implemented as recommended in Wang et al. doi 10.1016/j.combustflame.2017.08.005 (Eq. 1)
+        """
+        reduced_mass = self.get_reduced_mass(reverse)
+        sigma, epsilon = self.get_mean_sigma_and_epsilon(reverse)
+        Tr = temp * constants.kB * constants.Na / epsilon
+        reduced_coll_integral = 1.16145 * Tr ** (-0.14874) + 0.52487 * math.exp(-0.7732 * Tr) + 2.16178 * math.exp(
+            -2.437887 * Tr)
+        k_coll = (math.sqrt(8 * math.pi * constants.kB * temp * constants.Na / reduced_mass) * sigma ** 2
+                  * reduced_coll_integral * constants.Na)
+        return k_coll
+
+    def get_reduced_mass(self, reverse=False):
+        """
+        Returns the reduced mass of the reactants if reverse is ``False``
+        Returns the reduced mass of the products if reverse is ``True``
+        """
+        if reverse:
+            mass_list = [spc.molecule[0].getMolecularWeight() for spc in self.products]
+        else:
+            mass_list = [spc.molecule[0].getMolecularWeight() for spc in self.reactants]
+        reduced_mass = reduce((lambda x, y: x * y), mass_list) / sum(mass_list)
+        return reduced_mass
+
+    def get_mean_sigma_and_epsilon(self, reverse=False):
+        """
+        Calculates the collision diameter (sigma) using an arithmetic mean
+        Calculates the well depth (epsilon) using a geometric mean
+        If reverse is ``False`` the above is calculated for the reactants, otherwise for the products
+        """
+        sigmas = []
+        epsilons = []
+        if reverse:
+            for spc in self.products:
+                trans = spc.getTransportData()
+                sigmas.append(trans.sigma.value_si)
+                epsilons.append(trans.epsilon.value_si)
+                num_of_spcs = len(self.products)
```

Comment:
Move this out of the for loop. Otherwise, `num_of_spcs` is set to the same value over and over again

Reply:
done, thanks!